rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1173,11 +1173,13 @@ stream {
             {{ end }}
 
             {{/* By default use vhost as Host to upstream, but allow overrides */}}
+            {{ if not (eq $proxySetHeader "grpc_set_header") }}
             {{ if not (empty $location.UpstreamVhost) }}
             {{ $proxySetHeader }} Host                   "{{ $location.UpstreamVhost }}";
             {{ else }}
             {{ $proxySetHeader }} Host                   $best_http_host;
             {{ end }}
+            {{ end }}
```

Comment:
Is this the actual bug fix? Can you elaborate how?

Reply:
https://github.com/kubernetes/ingress-nginx/issues/3706#issuecomment-495157068

Reply:
@aledbf legend. Thanks!